app/controllers/batch_invitations_controller.rb

```diff
@@ -91,4 +96,17 @@ def grant_default_permissions(batch_invitation)
       batch_invitation.grant_permission(default_permission)
     end
   end
+
+  def batch_users_error_message(batch_user)
+    e = batch_user.errors.first
+
+    case e.attribute
+    when :email
+      "One or more emails were invalid"
+    when :name
+      "One or more names were invalid"
+    else
+      e.full_message
+    end
+  end
```

Comment:
In the commit note you say:

> This is a small improvement which avoids some of the complexities of what might be even more helpful error reports (e.g. listing one or all of the invalid email addresses).

I may well be missing something, but wouldn't it be _relatively_ easy to include the _first_ invalid email address by generating an error message along the following lines:

```ruby
"One or more emails were invalid, e.g. #{batch_user.email}"
```

I think this might make the error message more "actionable".

Reply:
Over the weekend I somehow convinced myself that including one example was more confusing for users than including none. So I removed something very close to this this morning :grimacing: 

I assume you're right

Reply:
> I assume you're right

Hmm - I wouldn't be too sure about that - it might be worth getting a 2nd opinion from Joseph or someone...?

Reply:
Joseph voted for keeping it without the example. After arguing in favour of some more complicated options!

I'll write up some ideas we had and add them to that other general improvements card (https://trello.com/c/kXMWrDu1/44-improve-batchinvitations)

Reply:
👍